samples/csharp/getting-started/Ranking_PersonalizedSort/README.md

```diff
@@ -0,0 +1,218 @@
+TrainTestData trainTestData = mlContext.Data.TrainTestSplit(labeledData, testFraction: 0.2, samplingKeyColumnName: nameof(HotelData.GroupId), seed: 1);
+IDataView trainData = trainTestData.TrainSet;
+IDataView testData = trainTestData.TestSet;
+
+// Save the test dataset to a file to make it faster to load in subsequent runs.
+using (var fileStream = File.Create(trainDatasetPath))
+{
+    mlContext.Data.SaveAsText(trainData, fileStream, separatorChar: ',', headerRow: true, schema: true);
+}
+
+// Save the train dataset to a file to make it faster to load in subsequent runs.
+using (var fileStream = File.Create(testDatasetPath))
+{
+    mlContext.Data.SaveAsText(testData, fileStream, separatorChar: ',', headerRow: true, schema: true);
+}
+`````
+
+### 3. Train Model
+This sample trains the model using the LightGbmRankingTrainer which relies on LightGBM LambdaRank.  The model requires the following input columns:
+
+* Group Id - Column that contains the group id for each data instance.  Data instances are contained in logical groupings representing all candidate results in a single query and each group has an identifier known as the group id.  In the case of the Expedia dataset, hotel search results are grouped by their corresponding query where the group id corresponds to the query or search id.  The input group id data type must be [key type](https://docs.microsoft.com/en-us/dotnet/api/microsoft.ml.data.keydataviewtype). 
+* Label - Column that contains the relevance label of each data instance where higher values indicate higher relevance.  The input label data type must be [key type](https://docs.microsoft.com/en-us/dotnet/api/microsoft.ml.data.keydataviewtype) or [Single](https://docs.microsoft.com/en-us/dotnet/api/system.single). 
+* Features - The columns that are influential in determining the relevance\rank of a data instance.  The input feature data must be a fixed size vector of type [Single](https://docs.microsoft.com/en-us/dotnet/api/system.single).
+
+When the trainer is set, **custom gains** (or relevance gains) are used to apply weights to each of the labeled relevance scores.  As described earlier in the sample, the potential relevance scores are {0, 1, 2} which directly correlates to relevance gains {0, 1, 5}.  This helps to ensure that the model places more emphasis on ranking hotel search results labeled with 2 (e.g. signifies the user purchased\booked the hotel) so that they are positioned higher when compared to results labeled with 0 or 1.    
+
+The following code is used to train the model:
+
+```CSharp
+const string FeaturesVectorName = "Features";
+
+// Load the training dataset.
+IDataView trainData = mlContext.Data.LoadFromTextFile<HotelData>(trainDatasetPath, separatorChar: ',', hasHeader: true);
+
+// Specify the columns to include in the feature input data.
+var featureCols = trainData.Schema.AsQueryable()
```

Comment:
Could this simply be:

```
var featureCols = new[] { nameof(HotelData.Price_USD), nameof(HotelData.Promotion_Flag), nameof(HotelData.Prop_Id), nameof(HotelData.Prop_Brand), nameof(HotelData.Prop_Review_Score)};
```

?

Reply:
I have since moved to a new dataset where there are 136 columns - as a result, this code is now: var featureCols = trainData.Schema.AsQueryable()
    .Select(s => s.Name)
    .Where(c =>
        c != nameof(SearchResultData.Label) &&
        c != nameof(SearchResultData.GroupId))
    .ToArray();

#resolved